packages/synapse-react-client/src/components/entity/metadata/EntityModal.integration.test.tsx

```diff
@@ -201,38 +179,119 @@ describe('EntityModal tests', () => {
         async (req, res, ctx) => {
           const response = cloneDeep(mockFileEntityBundle)
           ;(response.entity as FileEntity).isLatestVersion = true
-          response.permissions = {
-            canEdit: true,
-            // the rest don't matter
-            canAddChild: false,
-            canCertifiedUserAddChild: false,
-            canCertifiedUserEdit: false,
-            canChangePermissions: false,
-            canChangeSettings: false,
-            canDelete: false,
-            canDownload: false,
-            canEnableInheritance: false,
-            canModerate: false,
-            canMove: false,
-            canPublicRead: false,
-            canUpload: false,
-            canView: false,
-            isCertificationRequired: false,
-            isCertifiedUser: false,
-            isEntityOpenData: false,
-            ownerPrincipalId: 0,
-          }
+          response.permissions = PERMISSIONS_CAN_EDIT
           return res(ctx.status(200), ctx.json(response))
         },
       ),
     )
 
-    renderComponent({ initialTab: 'ANNOTATIONS' })
+    const { user } = renderComponent({
+      initialTab: 'ANNOTATIONS',
+      onEditModeChanged,
+    })
 
     const editButton = await screen.findByRole('button', { name: 'Edit' })
-    await userEvent.click(editButton)
+    await user.click(editButton)
+
+    expect(onEditModeChanged).toHaveBeenLastCalledWith(true)
 
     // Text that appears in the editor component if there's a schema
     await screen.findByText('Mock Annotation Editor', { exact: false })
   })
+
+  it('Cancelling out of edit mode using the cancel button triggers a warning before disabling edit mode', async () => {
```

Comment:
Some duplicate code in the 2 new tests that could be refactored to share common setup and verification